workflows/cloudify_system_workflows/tests/test_plugins.py

```diff
@@ -0,0 +1,123 @@
+########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import unittest
+from functools import wraps
+
+from mock import patch, MagicMock, PropertyMock, call
+
+from cloudify import decorators
+from cloudify.models_states import ExecutionState
+
+
+def mock_workflow_decorator(f=None, *_, **__):
+    if not f:
+        return lambda _f, *_, **__: mock_workflow_decorator(_f, *_, **__)
+
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        return f(*args, **kwargs)
+
+    return wrapper
+
+
+decorators.workflow = mock_workflow_decorator
+
+from cloudify_system_workflows.plugins import update as update_func  # noqa
+
+
+class TestPluginsUpdate(unittest.TestCase):
```

Comment:
maybe also test that you can not run this if another execution is already running in the system (not have to be also plugins update)

Reply:
Wouldn't that be the responsibility of the scheduler?

Reply:
no

Reply:
Waiting for Ido so see if I need to add that wait for running execs to finish.